docs/topics/testing/advanced.txt

```diff
@@ -307,6 +307,53 @@ running your tests, you can define test-only models in its ``models.py``
 file.
 
 
+Test-only models
+==========================================================
+
+Test-only models are models that are only created while running unit tests. This is particularly useful when writing :doc:`reusable applications </intro/reusable-apps>` that do not contain any models of their own.
+
+Consider the following structure::
+
+    polls/
+        migrations/
+            0001_initial.py
+            __init__.py
+        tests/
+            __init__.py
+            models.py
+            test_polls.py
+            ...
+        __init__.py
+        models.py
+        views.py
+        urls.py
+        ...
+
+The ``models.py`` file in the test module contains your test-only model definitions. It is important to leave the original ``models.py`` file in the app module even if it is empty.
+
+Create a migration for your test-models in the migrations module. You will need to adapt the migration to only execute during unit tests. You can do this by checking the database name::
```

Comment:
Yes 0002 will go into the migration chain after 0001 and before 0003. In the normal migration sequence, it will essentially be a noop ("no operation") because the real database will not start with 'test_'. In the migration sequence during unit tests however, the database will start with 'test_' and therefore migration 0002 will actually run and create tables for your test models.

I am not sure about what effect squashing migrations will have on this.